### DISCUSSION TITLE
Fast CDR exception deserializing message with Python

### DISCUSSION THREAD

**Discussion by JingyuQian** [foxglove/mcap#1238] [General]
Hello, I just encountered a weird error when deserializing mcap messages using python and can't figure out how to solve it. My steps were:
1. `colcon build` all the .msg files;
2. `source install/setup.bash`;
3. running the example given in [https://github.com/foxglove/mcap/blob/main/python/examples/ros2/py_mcap_demo/py_mcap_demo/reader.py](https://github.com/foxglove/mcap/blob/main/python/examples/ros2/py_mcap_demo/py_mcap_demo/reader.py)
4. get error message: `rclpy._rclpy_pybind11.RMWError: failed to deserialize ROS message: Fast CDR exception deserializing message of type pnc_msgs::msg::dds_::Decision_., at ./src/type_support_common.cpp:118`

I thought it was the .msg version mismatch (serialized mcap does not use the same version of .msg as built), so I tried deleting a field in one of the msg file and redo the process, in order to prove my guess (if true, then modified .msg should fail more deserializations). But this didn't pan out. 

I'm entirely new to this so any help is much appreciated.

**Comment by jtbandes** [comment]
Are you running this script inside of your ROS environment (i.e. the script is able to access the Decision msg definition?) How did you define the custom msg? Perhaps you can share a zip file of the package you are having trouble with along with a .mcap file so we can try it out.

**Reply by JingyuQian** [reply]
Thank you. I still think most likely it is a definition mismatch. I'm still working with the one who wrote the IDL definitions to figure out what the root cause is.
